tests/tests/test_image_io.py

```diff
@@ -1,63 +1,283 @@
     return np.tile(np.array([1, 2, 3, 4], dtype=np.int32), (4, 1))
 
 
 @pytest.fixture()
-def start_array(layer):
-    volume = np.dstack((layer, 2 * layer, 3 * layer, 4 * layer))
+def array_3d(array_2d):
+    """Create a 4x4x4 array of 32-bit integers"""
+    volume = np.stack((array_2d, 2 * array_2d, 3 * array_2d, 4 * array_2d))
     return volume
 
 
-def test_tiff_io(tmpdir, layer):
-    folder = str(tmpdir)
-    dest_path = os.path.join(folder, "layer.tiff")
-    tifffile.imwrite(dest_path, layer)
-    reloaded = tifffile.imread(dest_path)
-    assert (reloaded == layer).all()
-
-
-def test_to_tiffs(tmpdir, start_array):
-    folder = str(tmpdir)
-    save.to_tiffs(start_array, os.path.join(folder, "start_array"))
-    reloaded_array = load.load_from_folder(folder, 1, 1, 1)
-    assert (reloaded_array == start_array).all()
-
-
-def test_load_img_sequence(tmpdir, start_array):
-    folder = str(tmpdir.mkdir("sub"))
-    save.to_tiffs(start_array, os.path.join(folder, "start_array"))
-    img_sequence_file = tmpdir.join("imgs_file.txt")
-    img_sequence_file.write(
-        "\n".join(
-            [
-                os.path.join(folder, fname)
-                for fname in sorted(os.listdir(folder))
-            ]
-        )
+@pytest.fixture(params=["2D", "3D"])
+def image_array(request, array_2d, array_3d):
+    """Create both a 2D and 3D array of 32-bit integers"""
+    if request.param == "2D":
+        return array_2d
+    else:
+        return array_3d
+
+
+def write_tiff_sequence_with_txt_file(txt_path, image_array, shuffle=False):
+    """
+    Write an image array to a series of tiffs, and write a text file
+    containing all the tiff file paths in order (one per line).
+
+    The tiff sequence will be saved to a sub-folder inside the same folder
+    as the text file.
+
+    Parameters
+    ----------
+    txt_path : pathlib.Path
+        Filepath of text file to create
+    image_array : np.ndarray
+        Image to write as sequence of tiffs
+    shuffle : bool
+        Whether to shuffle the order of filepaths in the text file
+    """
+    directory = txt_path.parent
+
+    # Write tiff sequence to sub-folder
+    sub_dir = directory / "sub"
+    sub_dir.mkdir()
+    save.to_tiffs(image_array, str(sub_dir / "image_array"))
+
+    # Write txt file containing all tiff file paths (one per line)
+    tiff_paths = sorted(sub_dir.iterdir())
+    if shuffle:
+        random.Random(4).shuffle(tiff_paths)
+    txt_path.write_text(
+        "\n".join([str(sub_dir / fname) for fname in tiff_paths])
     )
-    reloaded_array = load.load_img_sequence(str(img_sequence_file), 1, 1, 1)
-    assert (reloaded_array == start_array).all()
 
 
-def test_to_nii(tmpdir, start_array):  # Also tests load_nii
-    folder = str(tmpdir)
-    nii_path = os.path.join(folder, "test_array.nii")
-    save.to_nii(start_array, nii_path)
-    assert (load.load_nii(nii_path).get_fdata() == start_array).all()
+def save_any(file_path, image_array):
```

Comment:
This is clever reducing of code duplication in our tests... I wonder whether it should actually go in the codebase itself :sweat_smile:

Reply:
..especially as we seem to have a `load_any` function?
Maybe you could open an issue for this?

Reply:
Sure! I'll open an issue for this

Reply:
Here's the issue: https://github.com/brainglobe/brainglobe-utils/issues/50